jicofo-common/src/main/kotlin/org/jitsi/jicofo/xmpp/muc/ChatRoomImpl.kt

```diff
@@ -362,78 +350,47 @@ class ChatRoomImpl(
     }
 
     override fun setPresenceExtension(extension: ExtensionElement, remove: Boolean) {
-        var presenceToSend: Presence? = null
-        synchronized(this) {
-            if (lastPresenceSent == null) {
-                logger.error("No presence packet obtained yet")
-                return
-            }
-            var presenceUpdated = false
+        val presenceToSend: Presence? = synchronized(this) {
+            lastPresenceSent?.let { presence ->
+                var presenceUpdated = false
 
-            // Remove old
-            val old = lastPresenceSent!!.getExtension(extension.qName)
-            if (old != null) {
-                lastPresenceSent!!.removeExtension(old)
-                presenceUpdated = true
-            }
-            if (!remove) {
-                // Add new
-                lastPresenceSent!!.addExtension(extension)
-                presenceUpdated = true
-            }
-            if (presenceUpdated) {
-                presenceToSend = lastPresenceSent!!.build()
+                presence.getExtension(extension.qName)?.let { old ->
+                    presence.removeExtension(old)
+                    presenceUpdated = true
+                }
+                if (!remove) {
+                    presence.addExtension(extension)
+                    presenceUpdated = true
+                }
+
+                if (presenceUpdated) presence.build() else null
+            } ?: run {
+                logger.error("No presence packet obtained yet")
+                null
             }
         }
-        if (presenceToSend != null) {
-            sendPresence(presenceToSend!!)
-        }
+        presenceToSend?.let { xmppProvider.xmppConnection.tryToSendStanza(it) }
     }
 
-    /**
-     * {@inheritDoc}
-     */
     override val presenceExtensions
-        @Synchronized
-        get() = lastPresenceSent?.extensions?.toList() ?: emptyList()
+        get() = synchronized(this) { lastPresenceSent?.extensions?.toList() ?: emptyList() }
 
-    /**
-     * {@inheritDoc}
-     */
     override fun modifyPresence(
         toRemove: Collection<ExtensionElement>,
         toAdd: Collection<ExtensionElement>
     ) {
-        var presenceToSend: Presence
-        synchronized(this) {
-            if (lastPresenceSent == null) {
+        val presenceToSend: Presence? = synchronized(this) {
```

Comment:
Does this need an explicit type declaration?

Reply:
Sorry, there were two instances and I only fixed one.